devito/symbolics/manipulation.py

```diff
@@ -61,7 +61,8 @@ def _uxreplace_handle(expr, args):
 @_uxreplace_handle.register(sympy.Add)
 def _(expr, args):
     if all(i.is_commutative for i in args):
-        _addsort(args)  # In-place sorting
+        _addsort(args)
```

Comment:
I know it's not new, but I remember looking at `addsort/mulsort` and was quite expensive (why moved from `subs` to `_subs` in part), do we actually need it there?

Reply:
yes we do, tried removing and all sorts of issues pop up. Here's the key problem: canonical ordering must be enforced upon expression lowering (e.g., CIRE needs it for max performance). Currently, canonical ordering is enforced during `lower_exprs` through this call to `uxreplace`. So if we want to pull it out of here, we need another pass to enforce the canonical ordering. I'm 100% in favor to it, but I'm not sure procrastinating the PR for this reason is the best way to go

Reply:
Ok, too bad, but I get it.